java/build.gradle

```diff
@@ -36,46 +36,104 @@ smallServices.each { Service svc ->
+        group 'deploy'
+        description "Deploy $svc.name services into the repo."
+        dependsOn "deploy${svc.name}Models"
+        outputs.upToDateWhen { false }
 
-        // service
-        def servicePath = "src/main/java/com/adyen/service"
-        def serviceSource = "services/$svc.id/${servicePath}"
-        from(layout.buildDirectory.dir("${serviceSource}/$svc.id")) {
-            include "*.java"
-            into "${servicePath}/${svc.id}"
+        // delete existing services
+        doFirst {
+            delete layout.projectDirectory.dir("repo/src/main/java/com/adyen/service/${serviceId}")
         }
 
-        // small service
-        from(layout.buildDirectory.dir(serviceSource)) {
+        // copy newly generated services
+        from layout.buildDirectory.dir("services/$svc.id/src/main/java/com/adyen/service/${serviceId}")
+        into layout.projectDirectory.dir("repo/src/main/java/com/adyen/service/" + serviceId)
+    }
+
+    // Copy serializers
+    def deploySerializers = tasks.register("deploy${svc.name}Serializers", Copy) {
+        group 'deploy'
+        description "Deploy $svc.name serializers into the repo."
+        dependsOn "deploy${svc.name}Services"
+        outputs.upToDateWhen { false }
+
+        // copy serializer JSON.java from service folder (if found) into model folder
+        def jsonJavaFileModelFolder = layout.buildDirectory.file("services/$svc.id/src/main/java/com/adyen/service/JSON.java")
+
+        from jsonJavaFileModelFolder
+        into layout.projectDirectory.file("repo/src/main/java/com/adyen/model/${serviceId}")
+
+        // copy serializer JSON.java from adyen folder (if found) into model folder
+        def jsonJavaFileAdyenFolder = layout.buildDirectory.file("services/$svc.id/src/main/java/com/adyen/JSON.java")
+
+        from jsonJavaFileAdyenFolder
+        into layout.projectDirectory.file("repo/src/main/java/com/adyen/model/${serviceId}")
+
+    }
+
+    def deploySmallServices = tasks.register("deploy${svc.name}SmallServices", Copy) {
+        group 'deploy'
+        description "Deploy $svc.name small services into the repo."
+        dependsOn "deploy${svc.name}Serializers"
+        outputs.upToDateWhen { false }
+
+        from(layout.buildDirectory.dir("services/$svc.id/src/main/java/com/adyen/service")) {
             include "*Single.java"
-            into servicePath
-            rename { svc.name + "Api.java" }
         }
+
+        into layout.projectDirectory.dir("repo/src/main/java/com/adyen/service").asFile
+
+        rename { svc.name + "Api.java" }
     }
 
-    tasks.named(svc.id) { dependsOn deploy }
+    tasks.named(svc.id) { dependsOn deployModels, deployServices, deploySerializers, deploySmallServices }
 }
+
+// Tests
+tasks.named('binlookup') {
+    doLast {
+        assert file("${layout.projectDirectory}/repo/src/main/java/com/adyen/model/binlookup/Amount.java").exists()
```

Comment:
Add comments in the TESTs to show which tests are targetting the 'small' services etc.